### ISSUE TITLE
Implement learningRadius parameter in Temporal Memory

### ISSUE THREAD

**Issue by rmccall84** [numenta/nupic#2110]
@param learningRadius      (int)   Radius around cell from which it can sample to form distal dendrite connections.

Default value = 2048

See: https://github.com/numenta/nupic/blob/master/nupic/research/temporal_memory.py#L538

**Comment by rcrowder** [comment]
@rmccall84 Hey, is this issue now closed via #2109 ?

**Comment by rmccall84** [comment]
@rcrowder No, this issue is not closed. The learning radius parameter was in Python TM, but it was not used nor did it have any effect so https://github.com/numenta/nupic/issues/2109 remove it. This issue suggests to implement learning radius completely. Thank you.

**Comment by cogmission** [comment]
Hi Guys,

Can you answer a question? What relationship does this have with the
boolean Global Inhibition setting? Are they related? If so, then how?

On Fri, Jun 26, 2015 at 11:41 AM, Ryan McCall <notifications@github.com>
wrote:

> @rcrowder <https://github.com/rcrowder> No, this issue is not closed. The
> learning radius parameter was in Python TM, but it was not used nor did it
> have any effect so #2109 <https://github.com/numenta/nupic/issues/2109>
> remove it. This issue suggests to implement learning radius completely.
> Thank you.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numenta/nupic/issues/2110#issuecomment-115749610>.
>



-- 
*With kind regards,*

David Ray
Java Solutions Architect

*Cortical.io <http://cortical.io/>*
Sponsor of:  HTM.java <https://github.com/numenta/htm.java>

d.ray@cortical.io
http://cortical.io

**Comment by rmccall84** [comment]
@cogmission I'm global inhibition concerns the SP inhibition step. Learning radius would define a neighborhood of cells about a learning cell. Distal synapse learning would only be allowed between some set of cells in this neighborhood and the learning cell. This is what I remember from the CLA days.

**Comment by breznak** [comment]
@rmccall84 :+1: 
could you implement it as a percentage, instead of an absolute value? `learningRadius`=(0,1] 
this would be part of #1343

**Comment by rmccall84** [comment]
@breznak Right now it looks like the selection is random amongst all cells. So this learning radius would go together with using a 2D topology. I'm not convinced a ratio for this parameter would be intuitive for users. What if I told you the learning radius is 0.00341796875 (ratio)? Alternatively we could say the learning radius is 7 (absolute). Just my thoughts, I don't feel too strongly about this =)
